src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -1920,6 +1930,10 @@ internal void Close()
             FullCleanup();
             LogMessages.ClosedPhysicalConnection(ConnectionLogger, Host, Port, Database, UserFacingConnectionString, Id);
         }
+        finally
+        {
+            _semaphore.Release();
```

Comment:
It doesn't look like we ever dispose the semaphore, which isn't nice.

Reply:
Fixed.